### ISSUE TITLE
Bleeding edge mesa crashes with steam

### ISSUE THREAD

**Issue by kaz0ku** [ValveSoftware/steam-for-linux#5095]
Hello, I've just installed Steam on my new PC but it crash at launch :(

#### Your system information
* Intel Core i5 7600K
* 16 GB DDR4
* MSI Z270 Tomawah
* SSD M.2 Corsair MP500 120GB
* HDD 2To (/home folder)
* AMD Radeon RX470 (Padoka PPA)
*
* Steam client version (build number or date):  can't start to check
* Distribution (e.g. Ubuntu): Ubuntu GNOME 17.04
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
Just after a clean install of Ubuntu GNOME 17.04 on my new PC I've just installed steam from terminal (just "apt install steam") after installation i start steam without error enter my login and password and just after just after "Updating user configuration" steam crash

Log from terminal 
https://gist.github.com/kaz0ku/429e1398d3a4b3dc82279acb9724188c

#### Steps for reproducing this issue:

1. Install steam from terminal apt-install steam
2. Start steam

**Comment by kisak-valve** [comment]
Hello @kaz0ku, can you give `LD_PRELOAD='/usr/$LIB/libstdc++.so.6 /usr/$LIB/libgpg-error.so' steam` a try in a terminal? (from https://github.com/ValveSoftware/steam-for-linux/issues/4451#issuecomment-220805021)

**Comment by kisak-valve** [comment]
Possibly this bleeding edge issue with mesa: https://bugs.freedesktop.org/show_bug.cgi?id=101876

If you are able, please try the latest point release of mesa like from xorg-edgers or pkppa.

**Comment by kaz0ku** [comment]
@kisak-valve Thanks, I've purged Padoka ppa and add the stable ppa from pkppa and now Steam can start correctly. So it seems to be a bug with the latest version of Mesa 17.2

**Comment by kisak-valve** [comment]
Thanks for testing, let's leave this issue open until the upstream issue is resolved.

**Comment by Zamundaaa** [comment]
Thank you very much. Using Padoka Stable Mesa PPA fixed this issue for me, too

**Comment by pbhj** [comment]
Me too - pkppa stable is working. Padoka's unstable and my original choice of oibaf both failed so I ppa-purged and went with pkppa (so far so good).

**Comment by kisak-valve** [comment]
Looks like a [fix](https://cgit.freedesktop.org/mesa/mesa/commit/?id=bbc29393d3beaf6344c7188547b4ff61b63946ae) has landed upstream, once Padoka and Oibaf has a build newer than this commit, please retest.

**Comment by pbhj** [comment]
Looks like [assuming these are full timestamps] [oibaf](https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers) has 170725 mesa and [Padoka's bleeding edge](https://launchpad.net/~paulo-miguel-dias/+archive/ubuntu/mesa/+packages) has 17072414 (just before the fix was committed), so I'll re-test oibaf and report back.

**Comment by veluria** [comment]
After installing the new beta, I can't get my Steam to start either. Arch with mesa 17.1.5-1. Is this even related? 

    assert_20170726184003_1.dmp[13216]: file ''/tmp/dumps/assert_20170726184003_1.dmp'', upload yes: ''CrashID=bp-abc113d5-8d8a-4f64-ba41-828d32170726''
    Assert( Assertion Failed: m_pInternalPipe->BRead failed ):../common/pipes.cpp:572

**Comment by kisak-valve** [comment]
Hello @veluria, that is unrelated to the issue being tracked in this issue report.

**Comment by mikedebian** [comment]
https://github.com/ValveSoftware/steam-for-linux/issues/5096

Is this not the bug that @veluria  is experiencing, that you closed in favor of this bug?

I am affected too, BTW. Same as @veluria. Running OpenSUSE Tumbleweed with MESA GIT.

```
crash_20170726191637_2.dmp[24260]: Uploading dump (out-of-process)
/tmp/dumps/crash_20170726191637_2.dmp
crash_20170726191637_2.dmp[24260]: Finished uploading minidump (out-of-process): success = yes
crash_20170726191637_2.dmp[24260]: response: Discarded=1
crash_20170726191637_2.dmp[24260]: file ''/tmp/dumps/crash_20170726191637_2.dmp'', upload yes: ''Discarded=1''
../common/pipes.cpp (572) : Assertion Failed: m_pInternalPipe->BRead failed
```

**Comment by JacobSvenningsen** [comment]
Ran into this problem about a week ago too, and can confirm the latest update provided by Padoka's bleeding-edge Mesa ppa no longer causes Steam to crash on launch

**Comment by kisak-valve** [comment]
Thanks for retesting. The fix is also in mesa 17.2.0-rc2, so I'm going to go ahead and close this out.
